datasets/narrativeqa_manual/narrativeqa_manual.py

```diff
@@ -0,0 +1,342 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""NarrativeQA Reading Comprehension Challenge"""
+
+from __future__ import absolute_import, division, print_function
+
+import csv
+import logging
+import os
+from os import listdir
+from os.path import isfile, join
+
+import datasets
+
+
+_CITATION = """\
+@article{kovcisky2018narrativeqa,
+  title={The narrativeqa reading comprehension challenge},
+  author={Ko{\v{c}}isk{\'y}, Tom{\'a}{\v{s}} and Schwarz, Jonathan and Blunsom, Phil and Dyer, Chris and Hermann, Karl Moritz and Melis, G{\'a}bor and Grefenstette, Edward},
+  journal={Transactions of the Association for Computational Linguistics},
+  volume={6},
+  pages={317--328},
+  year={2018},
+  publisher={MIT Press}
+}
+"""
+
+
+_DESCRIPTION = """\
+The Narrative QA Manual dataset is a reading comprehension \
+dataset, in which the reader must answer questions about stories \
+by reading entire books or movie scripts. \
+The QA tasks are designed so that successfully answering their questions \
+requires understanding the underlying narrative rather than \
+relying on shallow pattern matching or salience.\\
+THIS DATASET REQUIRES A MANUALLY DOWNLOADED FILE! \
+Because of a script in the original repository which downloads the stories from original URLs everytime, \
+The links are sometimes broken or invalid.  \
+Therefore, you need to manually download the stories for this dataset using the script provided by the authors \
+(https://github.com/deepmind/narrativeqa/blob/master/download_stories.sh). Running the shell script creates a folder named "tmp" \
+in the root directory and downloads the stories there. This folder containing the stories\
+can be used to load the dataset via `datasets.load_dataset("narrativeqa_manual", data_dir="<path/to/folder>").                """
+
+
+_HOMEPAGE = "https://deepmind.com/research/publications/narrativeqa-reading-comprehension-challenge"
+_LICENSE = "https://github.com/deepmind/narrativeqa/blob/master/LICENSE"
+
+
+# The HuggingFace dataset library don't host the datasets but only point to the original files
+# This can be an arbitrary nested dict/list of URLs (see below in `_split_generators` method)
+_URL = "https://github.com/deepmind/narrativeqa"
+_URLS = {
+    "documents": "https://raw.githubusercontent.com/deepmind/narrativeqa/master/documents.csv",
+    "summaries": "https://raw.githubusercontent.com/deepmind/narrativeqa/master/third_party/wikipedia/summaries.csv",
+    "qaps": "https://raw.githubusercontent.com/deepmind/narrativeqa/master/qaps.csv",
+}
+
+
+class NarrativeQaManual(datasets.GeneratorBasedBuilder):
```

Comment:
```suggestion
class NarrativeqaManual(datasets.GeneratorBasedBuilder):
```
This needs to be the camel case version of the dataset script name `narrativeqa_manual`